flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/source/KafkaSourceITCase.java

```diff
@@ -82,6 +82,57 @@ public void testBasicRead() throws Exception {
         executeAndVerify(env, stream);
     }
 
+    @Test
+    public void testParitionDiscoverySetting() throws Exception {
```

Comment:
It seems better to put this test into a new `KafkaSourceBuilderTest` class and separate it to a few methods testing one case each.

Reply:
It sounds pretty good